Tests/SwiftDocCTests/Infrastructure/DocumentationContext/DocumentationContextTests.swift

```diff
@@ -1984,6 +1984,47 @@ let expected = """
         XCTAssertEqual(unmatchedSidecarDiagnostic.severity, .warning)
     }
     
+    func testExtendingSymbolWithSpaceInName() throws {
+        let exampleDocumentation = Folder(name: "unit-test.docc", content: [
+            JSONFile(name: "ModuleName.symbols.json", content: makeSymbolGraph(
+                moduleName: "ModuleName",
+                symbols: [
+                    SymbolGraph.Symbol(
+                        identifier: .init(precise: "symbol-id", interfaceLanguage: "swift"),
+                        names: .init(title: "Symbol Name", navigator: nil, subHeading: nil, prose: nil),
+                        pathComponents: ["Symbol Name"],
+                        docComment: nil,
+                        accessLevel: .public,
+                        kind: .init(parsedIdentifier: .class, displayName: "Kind Display Name"),
+                        mixins: [:]
+                    )
+                ]
+            )),
+            
+            TextFile(name: "Extension.md", utf8Content: """
+            # ``Symbol Name``
+            
+            Extend a symbol with a space in its name.
+            """),
+            
+            TextFile(name: "Article.md", utf8Content: """
```

Comment:
Why add this article to the test? It doesn't look like you use it below for anything, and also the fix to the production code is for the links in the documentation extension titles, not links from inside of articles or in other places. Are those article links working in general? Or is this PR missing a fix for them as well? (and a check below to test links from articles to symbols with a space in the title)

Reply:
This article's content has a link to a symbol with a space in its name and this test verified that there are no problems raised. 

Even if it already was working, I wanted to add a test the verified that behavior.

Reply:
Ah ok thanks!